### PR TITLE
Mainmenu: Properly sort mods and games

### PR DESCRIPTION
**Pull request by SmallJoker**
This also removes trivial and unused pkgmgr functions
Fixes a bug caused by sorting in 2133fc8

## To do

This PR is Ready for Review.

## How to test

1. The removed functions must not be used anywhere
2. Creating a world works properly (correct game ID)
3. User-facing content listings are sorted (games, mods)

Bonus: The mods are now also sorted in the Settings tab.

### CODE REVIEW

File: builtin/mainmenu/dlg_create_world.lua [hunk 1]
```diff
@@ -111,7 +111,7 @@ local function create_world_formspec(dialogdata)
 	local game = pkgmgr.find_by_gameid(gameid)
 	if game == nil then
 		-- should never happen but just pick the first game
-		game = pkgmgr.get_game(1)
+		game = core.get_games()[1]
```
Comment (sfan5) [inline builtin/mainmenu/dlg_create_world.lua:114]: why not `pkgmgr.games[1]`?

File: src/script/lua_api/l_mainmenu.cpp [hunk 2]
```diff
@@ -540,8 +540,12 @@ int ModApiMainMenu::l_create_world(lua_State *L)
 			+ sanitizeDirName(name, "world_");
 
 	std::vector<SubgameSpec> games = getAvailableGames();
-	if (gameidx < 0 || gameidx >= (int) games.size()) {
-		lua_pushstring(L, "Invalid game index");
+	auto game_it = std::find_if(games.begin(), games.end(), [gameid] (const SubgameSpec &spec)
+	{
```
Comment (sfan5) [inline src/script/lua_api/l_mainmenu.cpp:544]: not sure if we have a standard for this but IMO this is less weird
```suggestion
	auto game_it = std::find_if(games.begin(), games.end(), [gameid] (const SubgameSpec &spec) {
```

File: builtin/mainmenu/dlg_create_world.lua (hunk 1 above)
Comment (SmallJoker) [reply builtin/mainmenu/dlg_create_world.lua:114]: because I forgot

**Review by sfan5 — APPROVED** [review]
Looks correct

File: builtin/mainmenu/pkgmgr.lua [hunk 3]
```diff
@@ -100,7 +100,15 @@ local function load_texture_packs(txtpath, retval)
...
 end
 
-function get_mods(path, virtual_path, retval, modpack)
+--modmanager implementation
+pkgmgr = {}
+
+--- Scans a directory recursively for mods and adds them to `listing`
+-- @param path         Absolute directory path to scan recursively
+-- @param virtual_path Prettified unique path (e.g. "mods", "mods/mt_modpack")
+-- @param listing      Input. Flat array to insert located mods and modpacks
```
Comment (rubenwardy) [inline builtin/mainmenu/pkgmgr.lua:109]: Shouldn't "Input" be "Output"?

Comment (SmallJoker) [reply builtin/mainmenu/pkgmgr.lua:109]: Yes and no. It's a mandatory table input which is then modified by reference

**Pull request merged**